src/main/comments.js

```diff
@@ -461,14 +463,30 @@ function printLeadingComment(path, options) {
   return parts;
 }
 
-function printTrailingComment(path, options) {
+function printTrailingComment(path, options, previousComment) {
   const comment = path.getValue();
   const printed = printComment(path, options);
 
   const { printer, originalText, locStart } = options;
-  const isBlock = printer.isBlockComment && printer.isBlockComment(comment);
 
-  if (hasNewline(originalText, locStart(comment), { backwards: true })) {
+  let previousCommentHasLineSuffix = false;
+
+  if (previousComment) {
+    const previousCommentDocType = getDocType(previousComment.doc);
+    if (
+      previousCommentDocType === DOC_TYPE_LINE_SUFFIX ||
+      (previousCommentDocType === DOC_TYPE_ARRAY &&
+        getDocType(previousComment.doc[0]) === DOC_TYPE_LINE_SUFFIX)
```

Comment:
Should we use `findInDoc`? The printed doc may change in future.

Reply:
The tests should save us. `findInDoc` is recursive. Seems like too much overhead.

As an alternative solution, what if we change the return type of this function to `{ doc, isBlock, hasLineSuffix }`

Reply:
Much better!